plugin/src/main/java/org/opensearch/ml/settings/MLCommonsSettings.java

```diff
@@ -177,4 +177,25 @@ private MLCommonsSettings() {}
     // Feature flag for enabling search processors for Retrieval Augmented Generation using OpenSearch and Remote Inference.
     public static final Setting<Boolean> ML_COMMONS_RAG_PIPELINE_FEATURE_ENABLED =
         GenerativeQAProcessorConstants.RAG_PIPELINE_FEATURE_ENABLED;
+
+    public static final Setting<Integer> ML_COMMONS_HTTP_CLIENT_CONNECTION_TIMEOUT_IN_MILLI_SECOND = Setting
```

Comment:
What if customer wants timeout = 60 ms for remote model A, and timeout = 80 ms for remote model B?

May be customer can provide these parameters during setting up the connector? We'll have some default values, but customer can provide their expected values also if they want?

Reply:
Why not include these three params in the request body of the "Predict" call to each LLM? If they are absent in the request, we can use the default values like these settings.

Reply:
This can be implemented but we'd better make these changes in a different PR, for this PR the target is to enable customer to set these parameters to avoid exceptions.

Reply:
@zane-neo So we have three options now
1. Add a cluster setting like what this PR does
2. Add these parameters to connector configuration, from @dhrubo-os 
3. User can set these parameters in request body.

For my understanding, we already have a lot of cluster settings, maybe let's not add more cluster settings? Seems adding to connector level is more flexible, and if we add to connector parameters, we can also support option3 by default.

Reply:
+1 on this.